### PR TITLE
add gh action CI system

### PR DESCRIPTION
**Pull request by camallen**
use GH actions for CI system and this would replace travis CI system

Totally up to you if you want to adopt GH Actions for CI

### CODE REVIEW

**Comment by mwalmsley** [comment]
I actually already have something very similar set up via Travis? See .travis.yml

**Comment by mwalmsley** [comment]
I think the action is looking too many "zoobot"'s deep

Obtaining file:///home/runner/work**/zoobot/zoobot/zoobot**
ERROR: file:///home/runner/work/zoobot/zoobot/zoobot does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.

**Comment by camallen** [comment]
> I actually already have something very similar set up via Travis? See .travis.yml
> 
> Edit - oh, I see, you're using this to test the build action

More that travis CI has become a troubling place to host your CI system lately. They were bought out and the company that bought them are squeezing all they can out of folks. The CI run times are long and queue times more so. Also they've had security issues lately as well so most folks are moving OSS systems away from them. 

This should co-locate the code and CI system to GH and speed up the CI times significantly as well (faster feedback is good)

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
```
Comment (camallen) [inline .github/workflows/python-package.yml:4]: Run Tests

Comment (camallen) [reply .github/workflows/python-package.yml:4]:
```suggestion
name: Run Tests
```

**Review by mwalmsley — APPROVED** [review]
LGTM. 
- Failing on 3.10 is expected, pytorch/tensorflow devs usually lag a little behind the latest python due to (I assume) engineering effort requiredd
- Added build requirement for 3.7, which many academics use

**Comment by mwalmsley** [comment]
Also, disabling Travis now via Travis UI

**Pull request merged**